pennylane/beta/tapes/tape.py

```diff
@@ -833,3 +835,105 @@ def data(self):
     @data.setter
     def data(self, params):
         self.set_parameters(params, trainable_only=False)
+
+    # ========================================================
+    # execution methods
+    # ========================================================
+
+    def execute(self, device, params=None):
```

Comment:
I have to admit, I'm a bit personally biased towards `execute` over `evaluate`. Partially because I find `execute` is more descriptive in that it will be actually calling the quantum device for the result.

Having a different term here will also make the final find-all/replace-all easier when we replace core!

But happy to change it back if everyone agrees